data/plugins/language_lua.lua

```diff
@@ -27,9 +27,53 @@ syntax.add {
     { pattern = "%.%.%.?",                   type = "operator" },
     { pattern = "[<>~=]=",                   type = "operator" },
     { pattern = "[%+%-=/%*%^%%#<>]",         type = "operator" },
-    { pattern = "[%a_][%w_]*()%s*%f[(\"'{]", type = {"function", "normal"} },
-    { pattern = "[%a_][%w_]*",               type = "symbol" },
+    -- Metamethods
+    { pattern = "__[%a][%w_]+",              type = "function" },
+    { pattern = "%.()__[%a][%w_]+",
+      type = { "normal", "function" }
+    },
+    -- Variable declaration
+    { pattern = "local()%s+()[%a_][%w_]+",
+      type = { "operator", "normal", "keyword2" }
+    },
+    -- Multiple variable declarations
+    { pattern = "[%a][%w_]*()%s*,%s*()[%a][%w_]*() *=",
+      type = { "keyword2", "normal", "keyword2", "operator" }
+    },
+    { pattern = "[%a][%w_]*()%s*,%s*()[%a][%w_]*()%s*,%s*()[%a][%w_]*()%s*=",
+      type = { "keyword2", "normal", "keyword2", "normal", "keyword2", "operator" }
+    },
+    { pattern = "[%a][%w_]*()%s*,%s*()[%a][%w_]*()%s*,%s*()[%a][%w_]*()%s*,%s*()[%a][%w_]*()%s*=",
+      type = { "keyword2", "normal", "keyword2", "normal", "keyword2", "normal", "keyword2", "operator" }
+    },
+    { pattern = "[%a][%w_]*()%s*,%s*()[%a][%w_]*()%s*,%s*()[%a][%w_]*()%s*,%s*()[%a][%w_]*()%s*,%s*()[%a][%w_]*%s*=",
+      type = { "keyword2", "normal", "keyword2", "normal", "keyword2", "normal", "keyword2", "normal", "keyword2", "operator", }
+    },
+    -- Function calls
+    { pattern = "[%a_][%w_]+()%s*%f[(\"'{]", type = {"function", "normal"} },
+    { pattern = "[%a_][%w_%.]+()%.()[%a_][%w_]+()%s*%f[(\"'{]",
+      type = { "normal", "normal", "function", "normal"} },
+    -- Sub fields
+    { pattern = "%.()[%a_][%w_]*%f[^%.%w=%(]",
```

Comment:
Why this `[^%.%w=%(]`? Did you want to avoid highlighting multiple sub-fields?

This breaks things like this:
![Schermata del 2022-06-04 06-24-06](https://user-images.githubusercontent.com/2798487/171982626-075ece4c-1559-4e00-b60a-abc38bc1c86f.png)

Reply:
yes, wanted to void matching all sub fields, will check that issue you found